client/src/components/common/Dashboard.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { message } from 'antd';
+import Addstory from '../AddStory/Addstory';
+import http from '../../services/httpService';
+import '../AddStory/Addstory.css';
+
+function Dashboard({ page = 1 }) {
+  const [story, setStory] = useState([]);
+  const [requestSent, setRequestSent] = useState(false);
+  const history = useHistory();
+
+  const fetchData = async () => {
+    try {
+      const { data } = await http.get(`/api/v1/story?page=${page}`);
+      return setStory(data);
+    } catch (error) {
+      if (error.response.status === 500) {
```

Comment:
Errors like this one in specific, should not be caught by this fetchData function, they should be caught by a generic handler that automatically handles 500 errors (And general system errors) and only leaves the component to handle errors that are related to it.

Reply:
Done